files/task_helper.ps1

```diff
@@ -0,0 +1,304 @@
+    {
+      $MightBeADate = $InputObject.get_InnerText() -as [DateTime]
+      ## Strings that are actually dates (*grumble* JSON is crap)
+      if ($MightBeADate -and $propbag."$Name" -eq $MightBeADate.ToString("G"))
+      {
+        return $MightBeADate
+      }
+      else
+      {
+        return $InputObject.get_InnerText()
+      }
+    }
+    'number'
+    {
+      $number = $InputObject.get_InnerText()
+      if ($number -eq ($number -as [int]))
+      {
+        return $number -as [int]
+      }
+      elseif ($number -eq ($number -as [double]))
+      {
+        return $number -as [double]
+      }
+      else
+      {
+        return $number -as [decimal]
+      }
+    }
+    'array' {
+      [object[]]$Items = $(
+        foreach( $item in $InputObject.GetEnumerator() )
+        {
+          Convert-Properties $item
+        }
+      )
+      return $Items
+    }
+    default { return $InputObject }
+  }
+}
+
+Function ConvertFrom-Json2
+{
+  [CmdletBinding()]
+  PARAM(
+    [Parameter(ValueFromPipeline=$true,Mandatory=$true,Position=1)]
+    [string]
+    $InputObject,
+
+    [Parameter(Mandatory=$true)]
+    [Type]
+    $Type,
+
+    [Switch]
+    $ForceType
+  )
+
+  $null = $PSBoundParameters.Remove("InputObject")
+  [Xml.XmlElement]$xml = (Convert-JsonToXml $InputObject).Root
+  if ($xml)
+  {
+    if ($xml.Objects)
+    {
+      $xml.Objects.Item.GetEnumerator() | ConvertFrom-Xml @PSBoundParameters
+    }
+    elseif ($xml.Item -and $xml.Item -isnot [System.Management.Automation.PSParameterizedProperty])
+    {
+      $xml.Item | ConvertFrom-Xml @PSBoundParameters
+    }
+    else
+    {
+      $xml | ConvertFrom-Xml @PSBoundParameters
+    }
+  }
+  else
+  {
+    Write-Error "Failed to parse JSON with JsonReader" -Debug:$false
+  }
+}
+
+function ConvertFrom-PSCustomObject
+{
+  PARAM(
+    [Parameter(ValueFromPipeline = $true)]
+    $InputObject
+  )
+
+  if ($null -eq $InputObject) { return $null }
+  if ($InputObject -is [System.Collections.IEnumerable] -and $InputObject -isnot [string])
+  {
+    $collection = @(
+      foreach ($object in $InputObject) { ConvertFrom-PSCustomObject $object }
+    )
+    $collection
+  }
+  elseif ($InputObject -is [System.Management.Automation.PSCustomObject])
+  {
+    $hash = @{}
+    foreach ($property in $InputObject.PSObject.Properties)
+    {
+      $hash[$property.Name] = ConvertFrom-PSCustomObject $property.Value
+    }
+    $hash
+  }
+  else
+  {
+    $InputObject
+  }
+}
+
+function Get-ContentAsJson
+{
+  [CmdletBinding()]
+  PARAM(
+    [Parameter(Mandatory = $true)]
+    $Text,
+
+    [Parameter(Mandatory = $false)]
+    [Text.Encoding]
+    $Encoding = [Text.Encoding]::UTF8
+  )
+
+  # using polyfill cmdlet on PS2, so pass type info
+  if ($PSVersionTable.PSVersion -lt [Version]'3.0')
+  {
+    $Text | ConvertFrom-Json2 -Type PSObject | ConvertFrom-PSCustomObject
+  }
+  else
+  {
+    $Text | ConvertFrom-Json | ConvertFrom-PSCustomObject
+  }
+}
+
+# import json, sys
```

Comment:
Do you still need this commented out block?

Reply:
That was there to serve as reference based on what's in the Python helpers... I can remove it from commit, yes